go/libraries/doltcore/merge/schema_merge_test.go

```diff
@@ -15,68 +15,569 @@
 package merge_test
 
 import (
+	"context"
+	"strings"
 	"testing"
 
-	"github.com/dolthub/dolt/go/libraries/doltcore/schema"
-
 	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/dolthub/dolt/go/libraries/doltcore/doltdb"
+	"github.com/dolthub/dolt/go/libraries/doltcore/dtestutils"
+	"github.com/dolthub/dolt/go/libraries/doltcore/merge"
+	"github.com/dolthub/dolt/go/libraries/doltcore/schema"
+	"github.com/dolthub/dolt/go/libraries/doltcore/sqle"
+	"github.com/dolthub/dolt/go/libraries/doltcore/sqle/sqlfmt"
+	"github.com/dolthub/dolt/go/libraries/doltcore/sqle/sqlutil"
+	"github.com/dolthub/dolt/go/libraries/doltcore/table/editor"
+	"github.com/dolthub/dolt/go/store/hash"
 )
 
+type schemaMergeTest struct {
+	name        string
+	ancestor    table
+	left, right table
+	merged      table
+	conflict    bool
+}
+
+type table struct {
+	ns   namedSchema
+	rows []sql.Row
+}
+
+type namedSchema struct {
+	name   string
+	sch    schema.Schema
+	create string
+}
+
 // TestMergeSchemas are schema merge integration tests from 2023
 func TestSchemaMerge(t *testing.T) {
-	t.Run("column mapping tests", func(t *testing.T) {
-		testSchemaMerge(t, colMappingTests)
+	t.Run("column add/drop tests", func(t *testing.T) {
+		testSchemaMerge(t, columnAddDropTests)
+	})
+	t.Run("column default tests", func(t *testing.T) {
+		testSchemaMerge(t, columnDefaultTests)
 	})
 	t.Run("column type change tests", func(t *testing.T) {
-		testSchemaMerge(t, colMappingTests)
+		testSchemaMerge(t, typeChangeTests)
+	})
+	t.Run("column reordering tests", func(t *testing.T) {
+		testSchemaMerge(t, columnReorderingTests)
 	})
 	t.Run("primary key change tests", func(t *testing.T) {
-		testSchemaMerge(t, colMappingTests)
+		testSchemaMerge(t, keyChangeTests)
+	})
+	t.Run("secondary index tests", func(t *testing.T) {
+		testSchemaMerge(t, secondaryIndexTests)
+	})
```

Comment:
(minor) seems like we want to run all the test suite with multiple secondary indexes, to make sure all those cases work and that we exercise the multiple secondary index codepaths. I tried out the `validateIndexes` function you pointed me at last Friday and it triggered as expected, so I'm wondering if we can just bake multiple secondary indexes into all the tests instead of having a separate section of test cases for them.

Reply:
I think this makes a lot of sense. We can run multiple iterations of the test to validate the merge works with various secodary indexes. I'll add this in a follow up PR